src/Data/ObjectMap.purs

```diff
@@ -1,11 +1,12 @@
 module Data.ObjectMap
-  ( ObjectMap
+  ( ObjectMap(..)
```

Comment:
Ok, I see. You need the constructor for `unsafeFreeze` in the other module. I would prefer to find another solution here. I think hiding the implementation is important.

- I am not so experienced with `ST`, so I ask: do we need `unsafeFreeze`?
- If we need it, can we implement it without the constructor?
- Otherwise I'd go for the option to introduce an `.Internal` module that exposes the constructor and the main module would still hide it.

Reply:
Yes, I understand your point.

`unsafeFreeze` (or `freeze` that does a copy) is the only way to convert a mutable ObjectMap to a Map and thus to "leave" the ST monad.
I don't see how to implement it without the constructor so I guess intoducing an `Internal` module is the best solution.
If you agree, I will do that and update the README.